File: backend-nestjs/src/entities/sitemap-custom-url.entity.ts
import { Column, CreateDateColumn, Entity, Index, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';

export enum SitemapLastmodMode {
  NOW = 'now',
  FIXED_DATE = 'fixed_date',
  NONE = 'none',
}

@Entity({ name: 'sitemap_custom_urls' })
@Index(['tenantId'])
export class SitemapCustomUrlEntity {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column({ name: 'tenant_id', type: 'varchar', length: 36 })
  tenantId!: string;

  @Column({ name: 'path_or_url', type: 'varchar', length: 1024 })
  pathOrUrl!: string;

  @Column({ type: 'boolean', default: true })
  enabled!: boolean;

  @Column({
    name: 'lastmod_mode',
    type: 'enum',
    enum: SitemapLastmodMode,
    default: SitemapLastmodMode.NONE,
  })
  lastmodMode!: SitemapLastmodMode;

  @Column({ name: 'lastmod_value', type: 'timestamptz', nullable: true })
  lastmodValue!: Date | null;

  @Column({ type: 'varchar', length: 16, nullable: true })
  changefreq!: string | null;

  @Column({
    type: 'decimal',
    precision: 2,
    scale: 1,
    nullable: true,
  })
  priority!: string | null;

  @Column({ type: 'varchar', length: 512, nullable: true })
  notes!: string | null;

  @CreateDateColumn({ name: 'created_at', type: 'timestamptz' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'updated_at', type: 'timestamptz' })
  updatedAt!: Date;
}

File: backend/src/main/java/com/contentplatform/backend/infrastructure/jpa/entity/SitemapCustomUrlEntity.java
package com.contentplatform.backend.infrastructure.jpa.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

@Entity
@Table(name = "sitemap_custom_urls")
public class SitemapCustomUrlEntity {
    @Id
    @Column(name = "id", nullable = false, length = 36)
    private String id;

    @Column(name = "tenant_id", nullable = false, length = 36)
    private String tenantId;

    @Column(name = "path_or_url", nullable = false, length = 1024)
    private String pathOrUrl;

    @Column(name = "enabled", nullable = false)
    private boolean enabled;

    @Column(name = "lastmod_mode", nullable = false, length = 16)
    private String lastmodMode;

    @Column(name = "lastmod_value")
    private Instant lastmodValue;

    @Column(name = "changefreq", length = 16)
    private String changefreq;

    @Column(name = "priority", precision = 2, scale = 1)
    private Double priority;

    @Column(name = "notes", length = 512)
    private String notes;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    protected SitemapCustomUrlEntity() {
    }

    public SitemapCustomUrlEntity(String id,
                                  String tenantId,
                                  String pathOrUrl,
                                  boolean enabled,
                                  String lastmodMode,
                                  Instant lastmodValue,
                                  String changefreq,
                                  Double priority,
                                  String notes,
                                  Instant createdAt,
                                  Instant updatedAt) {
        this.id = id;
        this.tenantId = tenantId;
        this.pathOrUrl = pathOrUrl;
        this.enabled = enabled;
        this.lastmodMode = lastmodMode;
        this.lastmodValue = lastmodValue;
        this.changefreq = changefreq;
        this.priority = priority;
        this.notes = notes;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public String getId() {
        return id;
    }

    public String getTenantId() {
        return tenantId;
    }

    public String getPathOrUrl() {
        return pathOrUrl;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public String getLastmodMode() {
        return lastmodMode;
    }

    public Instant getLastmodValue() {
        return lastmodValue;
    }

    public String getChangefreq() {
        return changefreq;
    }

    public Double getPriority() {
        return priority;
    }

    public String getNotes() {
        return notes;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }
}

